### DISCUSSION TITLE
Better Strings - Java String Interpolation

### DISCUSSION THREAD

**Discussion by MarcinOrlowski** [logisim-evolution/logisim-evolution#1106] [Dev talk]
There's nice javac plugin called [Better Strings](https://github.com/antkorwin/better-strings) that would let us drop explicit string concatenation in favor of variable interpolation, so instead of

```java
var a = 3;
var b = 4;
System.out.println(a + " + " + b + " = " + (a + b));
```

you would be able to write as in other languages:

```java
var a = 3;
var b = 4;
System.out.println("${a} + ${b} = ${a+b}");
```

My tests shows it works fine, but take your time and speak up for or against.

**Comment by maehne** [comment]
I am not in favour of using non-standard language extensions to Java. I would prefer that we stick to only those language constructs, which have been voted in through a JEP and have been implemented in official JDK releases and are fully supported, i.e., have no preview status. This ensures maximum compatibility and makes it easier for potential new contributors to Logisim-evolution. IMHO, the recent refactorings driven especially by @MarcinOrlowski have already helped a lot to improve the readability of our code base.
